stdlib/src/builtin/simd.mojo

```diff
@@ -1879,6 +1880,63 @@ struct SIMD[type: DType, size: Int](
 
         return bitcast[_integral_type_of[type](), size](self).cast[int_dtype]()
 
+    @staticmethod
+    fn from_bytes[
+        big_endian: Bool = False
+    ](bytes: Span[Byte]) raises -> Scalar[type]:
+        """Converts a byte array to an integer.
+
+        Args:
+            bytes: The byte array to convert.
+
+        Parameters:
+            big_endian: Whether the byte array is big-endian.
+
+        Returns:
+            The integer value.
+        """
+        if type.sizeof() != len(bytes):
+            raise Error("Byte array size does not match the integer size.")
+
+        var ptr: UnsafePointer[Scalar[type]] = bytes.unsafe_ptr().bitcast[
+            Scalar[type]
+        ]()
+        var value = ptr[]
+
+        @parameter
+        if is_big_endian() and not big_endian:
+            value = byte_swap(value)
+        elif not is_big_endian() and big_endian:
+            value = byte_swap(value)
+        return value
+
+    fn as_bytes[big_endian: Bool = False](self) -> List[Byte]:
```

Comment:
**thought**: maybe this should return `InlineArray[Byte, type.sizeof()]`? That way we don't allocate

Reply:
Good suggestion. I tried it in this commit: https://github.com/msaelices/mojo/commit/8caad4136b92231e9541b94076ab618871c2148c

Unfortunately, it's breaking the test with a weird compiler error, which is a Mojo bug IMO:
![image](https://github.com/user-attachments/assets/3bdfb67f-3f7c-4d6f-bafb-2f51eedfc992)

@lsh, do you have any ideas? If not, we can revert the commit, add a TODO line for future improvement, and use the `List[Byte]` in the meantime.

Reply:
It's not a compiler bug, it says so in the error. You need to initialize the array with a value to fill the whole array or pass it an argument to unsafely initialize it. Set it to 0 and it should be good to go.

Reply:
Sorry, the `call expansion failed` confused me. I need to read the errors better. 

Fixed here: https://github.com/msaelices/mojo/commit/dd2981411720f5dcf0cb7acf36b6200e4c3d4f11

Thanks